FRENCH/src/ch06-02-match.md

```diff
@@ -0,0 +1,673 @@
+
+```rust,ignore
+None => None,
+```
+
+<!--
+The `Some(5)` value doesn’t match the pattern `None`, so we continue to the
+next arm.
+-->
+
+La valeur `Some(5)` ne correspond pas au motif `None`, donc nous continuons à la
+prochaine branche.
+
+```rust,ignore
+Some(i) => Some(i + 1),
+```
+
+<!--
+Does `Some(5)` match `Some(i)`? Why yes it does! We have the same variant. The
+`i` binds to the value contained in `Some`, so `i` takes the value `5`. The
+code in the match arm is then executed, so we add 1 to the value of `i` and
+create a new `Some` value with our total `6` inside.
+-->
+
+Est-ce que `Some(5)` correspond au `Some(i)` ? Oui, pourquoi pas ? Nous avons la
+même variante. Le `i` va prendre la valeur contenue dans le `Some`, donc `i`
+prend la valeur `5`. La code dans la branche du `match` est exécuté, donc nous
+ajoutons 1 à la valeur de `i` et nous créons une nouvelle valeur `Some` avec
+notre résultat `6` à l'intérieur.
+
+<!--
+Now let’s consider the second call of `plus_one` in Listing 6-5, where `x` is
+`None`. We enter the `match` and compare to the first arm.
+-->
+
+Maintenant, regardons le second appel à `plus_un` dans l'encart 6-5, où `x` vaut
+`None`. Nous entrons dans le `match` et nous le comparons à la première branche.
+
+```rust,ignore
+None => None,
+```
+
+<!--
+It matches! There’s no value to add to, so the program stops and returns the
+`None` value on the right side of `=>`. Because the first arm matched, no other
+arms are compared.
+-->
+
+Cela correspond ! Il n'y a pas de valeur à additionner, donc le programmme
+s'arrête et retourne la valeur `None` qui est dans le côté droit du `=>`. Comme
+la première branche correspond, les autres branches ne sont pas comparées.
+
+<!--
+Combining `match` and enums is useful in many situations. You’ll see this
+pattern a lot in Rust code: `match` against an enum, bind a variable to the
+data inside, and then execute code based on it. It’s a bit tricky at first, but
+once you get used to it, you’ll wish you had it in all languages. It’s
+consistently a user favorite.
+-->
+
+La combinaison de `match` et des énumérations est utile dans de nombreuses
+situations. Vous allez revoir de nombreuses fois ce schéma dans du code Rust :
+utiliser `match` sur une énumération, récupérer la valeur qu'elle renferme, et
+exécuter du code en fonction de sa valeur. C'est un peu délicat au début, mais
+une fois que vous vous y êtes habitué, vous regretterez de ne pas l'avoir dans
+les autres langages. Cela devient toujours l'outil préféré de ses utilisateurs.
+
+<!--
+### Matches Are Exhaustive
+-->
+
+### Les `match` sont toujours exhaustifs
+
+<!--
+There’s one other aspect of `match` we need to discuss. Consider this version
+of our `plus_one` function that has a bug and won’t compile:
+-->
+
+Il y a un autre point de `match` que nous devons aborder. Admettons que cette
+version de notre fonction `plus_un` a un bogue et ne va pas se compiler :
```

Comment:
We have made the choice to translate a maximum of words in french, so all the previous and next pages uses already `bogue`.
By the way, I agree with you, a hudge majority of people uses this anglicism, but I'm the one who use this word in personal et professional life.
I think for this case if we chose one or another, it will not be difficult for our readers to understand the meaning of this word :smile:
If this comes as a shock to you, I'm disposed to make this change but it involves far more work than let it like that :wink:

Reply:
We should decide whether to change it or not, without looking at how much of an effort it is to make the change. For the chapters already released to the `french-release` branch, we just have to take any text editor and `find and replace in project` 😄 . For the pending PRs, the change can be made during proofreading.

@joeljpresent What is your opinion about this one?

Reply:
Whilst *bug* is more common than *bogue*, it is common in French translations of documentations to use "pure" French equivalents rather than anglicisms.

Besides, using *bug* rather than *bogue* has a drawback: how do you translate the verb *to bug*?
*Buger*, *bugger*, *buguer*, *bugguer*, *beuguer*...?
Whereas using *bogue* makes the translation straightforward: *boguer*, as well as the translation of *debugger* to *débogueur*.
In that sense, keeping *bogue* is beneficial.

Reply:
Right.
@Jimskapt this review can be marked resolved, so we are able to merge the PR and decide whether to replace `bogue` with `bug` in a future discussion/PR

Reply:
nice, I'll do changes then merging tomorrow